proposals/features-composite-dependencies.md

```diff
@@ -0,0 +1,117 @@
+
+Composing Features should provide an alternative to existing community solutions, code duplication, and "hacky" means of installing a dependent Feature before another.
+
+## Definitions
+
+- `Standalone Feature` - The existing dev container Feature format, as outlined in the [spec](https://containers.dev/implementors/features/), and defined by a `devcontainer-feature.json` metadata file. A Feature that is self-contained, shareable unit of installation code and development container configuration.
+- `Composite Feature` - A Feature that depends on >= 1 other Features, defined by a `devcontainer-feature.composite.json` metadata file.
+
+## Existing community solutions
+
+### @danielBraun89 + '@devcontainers-contrib'
+
+This community repository containing 100+ Features provides a custom solution for dependencies, introducing an additional `feature-definition.json` file - superset of the `devcontainer-feature.json` with a [`dependencies` object](https://github.com/devcontainers-contrib/features/blob/db45f607e733f3d560f6527d89b6a9a85b3b806c/feature_definitions/elixir-asdf/feature-definition.json#L29-L50).  Their [custom CLI has a tool named `featmake`](https://github.com/devcontainers-contrib/cli/blob/main/resources/featmake/featmake.sh#L145-L149) that will use the `oras` tool to pull and execute the `install.sh` of the given Feature.  This strategy doesn't appear to merge in the other dev container configuration properties that a Feature may declare.
+
+### Direct curl
+
+We've seen instances where users directly `curl` a Feature's `install.sh` script to `bash`.  This strategy also doesn't merge in the other dev container configuration properties that a Feature may declare.
+
+### Additional inspiration
+
+Inspiration was taken from [this spec issue on the topic](https://github.com/devcontainers/spec/issues/109), the repositories listed above, [the buildpack specification](https://docs.cloudfoundry.org/buildpacks/understand-buildpacks.html), and [VS Code extension packs](https://code.visualstudio.com/blogs/2017/03/07/extension-pack-roundup).
+
+## Specification
+
+Introduce a new file type `devcontainer-feature.composite.json` with the following properties.
+
+| Property | Type | Description |
+|----------|------|-------------|
+| `id` | `string` | The ID of the Feature.  This follows the same semantics of the `id` property in the `devcontainer-feature.json` file. |
+| `version` | `string` | The version of the Feature.  This follows the same semantics of the `version` property in the `devcontainer-feature.json` file. |
+| `features` | `array` | An array of objects (in installation order) that define the Feature(s) that compose this Feature. |
+| `features.id` | `string` | The ID of the Feature that this Feature depends on. |
```

Comment:
Can these be from a different repository/namespace?

Reply:
Yes, I see no reason to restrict if this proposal is followed.